### PR TITLE
Add Integration test for TPM driver

### PR DESCRIPTION
**Pull request by praveen-pk**

### CODE REVIEW

File: resources/Dockerfile
Comment (likebreath) [inline resources/Dockerfile:137]: Can you please consolidate the dependencies to the "apt-get" step above from line 58 and also ensure no duplicate packages?

File: tests/integration.rs
Comment (likebreath) [inline tests/integration.rs:6780]: If you are planning to test the tpm module in the guest like what's described in the [tpm.md](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/tpm.md#testing), you can refer this document to include the `tpm2-tss` package in our custom guest image: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/custom-image.md.

File: resources/Dockerfile
Comment (praveen-pk) [reply resources/Dockerfile]: Done!

File: tests/integration.rs
Comment (rbradford) [inline tests/integration.rs:226]: nit: vertical whitespace

File: tests/integration.rs
Comment (rbradford) [inline tests/integration.rs:229]: You can keep everything as a `PathBuf` and convert to string only when needed.

File: tests/integration.rs
Comment (liuw) [inline tests/integration.rs:6754]: Why not? What's stopping this from working on MSHV?

File: tests/integration.rs
Comment (liuw) [inline tests/integration.rs:6771]: Please add a space after `//` and preferably capitalize `start`.

File: tests/integration.rs
Comment (liuw) [inline tests/integration.rs:6778]: This command alone will not exercise the swtpm backend in CH. I think the sequence needs to be more sophisticated.

File: tests/integration.rs
Comment (sboeuf) [inline tests/integration.rs:6753]: And what's stopping this from working on `aarch64`?

File: tests/integration.rs
Comment (sboeuf) [inline tests/integration.rs:6773]: Do you need these 2 seconds? I'm asking because either you really need them and I don't think 2 seconds is safe enough to make it reliable, or you don't need them because Cloud Hypervisor will retry multiple times to connect to the backend for a specific amount of time.

File: tests/integration.rs
Comment (rbradford) [reply tests/integration.rs:6780]: @praveen-pk Since the storage bucket is access controlled I can create that updated image for you.

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs:6778]: I agree. As mentioned in a previous comment, I need a customized guest image to run additional checks.

I wanted to get a basic test merged without being blocked by custom guest image. So, I started with this one.

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs:6773]: I noticed some failures during early development and left this in. I checked this today with 100 runs. They are all pass without a sleep in between. I will drop this in my next update. We can revisit this if we see failures in the future.

Comment (sboeuf) [reply tests/integration.rs:6773]: Well looking at the code, it seems you're lucky and possibly under high load, this is gonna fail. `Tpm::new()` -> `Emulator::new()` -> `SocketDev::init(path)` -> `Socketdev::connect(path)` -> `UnixStream::connect(path)` clearly shows there's no retry mechanism implemented, therefore you should give the `swtpm` daemon enough time to run and listen on the socket path before you run the Cloud Hypervisor binary.
I think it's better to use at least 10 seconds.

Comment (praveen-pk) [reply tests/integration.rs:6773]: Gotcha. I thought there was retry login in TPM module for 30 secs. Seems like I dropped the retry logic in TPM module before the final merge.
I will keep the 10 sec delay as you suggested.

I ran some tests last night and did see a failure where cloud-hypervisor failed to connect to swtpm socket.

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs]: done

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs:6762]: done

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs:237]: done

File: tests/integration.rs
Comment (praveen-pk) [reply tests/integration.rs:227]: done

File: tests/integration.rs
Comment (likebreath) [reply tests/integration.rs:6788]: @praveen-pk Rob's PR #5117 just landed and it updated jammy image with `tpm2-tools` installed. You can use this updated image in your integration test by simply switching to `UbuntuDiskConfig::new(JAMMY_IMAGE_NAME.to_string())`.

Comment (praveen-pk) [reply tests/integration.rs:6788]: Thanks @rbradford! I added 2 more tests while using the customized jammy image.

**Comment by praveen-pk** [comment]
updated the date in dev_cli.sh to today's date. I am hoping this PR can be merged today.

**Review by likebreath — APPROVED** [review]

**Pull request merged**
